src/utils.c

```diff
@@ -513,13 +513,16 @@ ts_type_is_int8_binary_compatible(Oid sourcetype)
  * Create a fresh struct pointer that will contain copied contents of the tuple.
  * Note that this function uses GETSTRUCT, which will not work correctly for tuple types
  * that might have variable lengths.
+ * Also note that the function assumes no NULLs in the tuple.
  */
 void *
 ts_create_struct_from_tuple(HeapTuple tuple, MemoryContext mctx, size_t alloc_size,
 							size_t copy_size)
 {
 	void *struct_ptr = MemoryContextAllocZero(mctx, alloc_size);
 
+	/* Make sure the function is not used when the tuple contains NULLs */
+	Assert(copy_size == tuple->t_len - tuple->t_data->t_hoff);
```

Comment:
This is not related to the rest of the PR and I am happy to remove it. However, I initially assumed I could use the function even when there are NULL columns and took waay to long to find and fix the bug. A check like this would save a lot of time.

Reply:
This is great